Userland/Libraries/LibGUI/VimEditingEngine.cpp

```diff
@@ -1403,4 +1404,188 @@ void VimEditingEngine::move_to_next_empty_lines_block()
     m_editor->set_cursor(new_cursor);
 };
 
+void VimEditingEngine::initialize_commands()
```

Comment:
We generally don't use strings to specify flags/modes. How about making an enum class like this:
```cpp
enum class VimMode {
    Normal,
    Insert,
    Visual,
    Command,
    // ...
};
```
and using `AK/EnumBits.h` to add bitwise operators?

Additionally, using strings to represent a keycode sequence is not very nice, even if it is terse. So perhaps that can become a Vector of KeyCodes passed to the `VimCommand` constructor.

With those in mind, a command can be written as:
```cpp
m_commands.append(VimCommand {
    { Key_V },
    VimMode::Normal,
    [&] {
        switch_to_visual_mode();
    }
});
```